python/sedona/sql/st_functions.py

```diff
@@ -1422,3 +1423,14 @@ def ST_HausdorffDistance(g1: ColumnOrName, g2: ColumnOrName, densityFrac: Option
     args = (g1, g2, densityFrac)
     return _call_st_function("ST_HausdorffDistance", args)
 
+@validate_argument_types
+def ST_CoordDim(geometry: ColumnOrName) -> Column:
```

Comment:
This function is not tested. Please read the `test_dataframe_api.py` in this PR: https://github.com/apache/sedona/commit/b6702d1adacb98d83cc470e9b24877ab7785cdf8#diff-ab6aae6fd177bcb406674e400a6370cf621d76c5d2de30678a3d521bb9b5a5cc